scico/linop/_linop.py

```diff
@@ -169,7 +169,7 @@ class Diagonal(LinearOperator):
 
     def __init__(
         self,
-        diagonal: Union[Array, BlockArray],
```

Comment:
@Michael-T-McCann: I think this was originally added by you. Please conform that the change from `Array` to `JaxArray` (which silences the warning) is indeed correct.

Reply:
I think in this case, changing Array to JaxArray means that the developer using the linop has to create the jax array beforehand. If we leave it as Array, it can take a numpy array or anything and convert it inside it using the ensure_on_device method. 

Essentially, the warning means that there is a conversion to JaxArray from np.array and I'm not sure if we want to allow the linop to be created using anything other than device/block arrays. To my knowledge, it doesn't affect anything code-wise due to the ensure_on_device but it brings attention to if we want to restrict it to that type.

Reply:
That's a reasonable argument, but it seems that all the other LinOps use `JaxArray` rather than `Array`. In fact, although `Array` is indeed defined in `scico.typing`, from a quick scan I couldn't find a single instance of it actually being used.

Reply:
Given that other code uses `JaxArray`, let's go with `JaxArray` here.

As a general principle, I believe we are _not_ going out of our way to be interoperable with numpy.